Source/NSOutlineView.m

```diff
@@ -2273,19 +2155,24 @@ - (void) _noteNumberOfRowsChangedBelowItem: (id)item by: (NSInteger)numItems
 - (NSCell *) preparedCellAtColumn: (NSInteger)columnIndex row: (NSInteger)rowIndex
 {
   NSCell *cell = nil;
-  NSTableColumn *tb = [_tableColumns objectAtIndex: columnIndex];
 
-  if ([_delegate respondsToSelector:
-        @selector(outlineView:dataCellForTableColumn:item:)])
+  if (_viewBased)
```

Comment:
You put the old code into a `_viewBased` condition. This look completely wrong, can you explain it?

Reply:
This method is called from several places in the code in NSTableView and NSOutlineView.   If it returns nil, then the code from which it is called acts appropriately.  No data cell should be returned when a table/outline is view-based.